src/partr.c

```diff
@@ -505,9 +505,14 @@ JL_DLLEXPORT jl_task_t *jl_task_get_next(jl_value_t *getsticky)
             start_cycles = 0;
         }
         else {
+#ifndef JL_HAVE_ASYNCIFY
             // maybe check the kernel for new messages too
             if (jl_atomic_load(&jl_uv_n_waiters) == 0)
                 jl_process_events(jl_global_event_loop());
+#else
+            // Yield back to browser event loop
+            return ptls->root_task;
```

Comment:
this seems unfortunate, since it appears to reflect that `root_task` means something different for jsvm and everyone else. it's also adding code to a part of the code that should be unreachable from wasm currently (it's only active in a threads-enabled build).

Reply:
Well, root_task is the one that we are in when we start up the runtime and happens to be the one that runs the browser event loop. I don't think this is too different from the meaning of root_task for other platforms.